.github/workflows/publish-to-pypi.yml

```diff
@@ -48,13 +47,15 @@ jobs:
         twine check dist/*
 
     - name: 📤 Publish source to Test PyPI
+      if: env.TWINE_PASSWORD != null
```

Comment:
Shouldn't this be `if: secrets.PYPI_TEST_PASSWORD != null`?  Maybe I'm misreading it?

Reply:
That does not work. It throws an error because `secrets` does not exist. Passing the value as an envvar allows to work around it.

Reply:
Is there an equivalent of `exists(secrets) && is_key(secretes, 'PYPI_TEST_PASSWORD')`?

Reply:
https://github.community/t/how-can-i-test-if-secrets-are-available-in-an-action/17911/7

Looks like we need to short circuit in the step itself?

Reply:
That's what we are doing. This is "using the env context in a step’s if conditional".

Reply:
> Is there an equivalent of `exists(secrets) && is_key(secretes, 'PYPI_TEST_PASSWORD')`?

There is not. I also tried `contains(secrets, 'PYPI_TEST_PASSWORD')`. See https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions

Reply:
For clarity, the `if` should be below the env, because the `if` acts on the result of the `env`, yes?

Reply:
Yes. The order of the fields is independent from how they are evaluated. Yet, I'll reorder it for clarity.

Reply:
Done.